File: lulu/a_b_problem.py
class Solution:
    """
    @param a: The first integer
    @param b: The second integer
    @return:  The sum of a and b
    """
    # Actually both of these methods fail to work in Python
    # because Python supports infinite integer
    # Non-recursive
    def aplusb(self, a, b):
        # write your code here, try to do it without arithmetic operators.
        while b != 0:
            carry = a & b
            a = a ^ b
            b = carry << 1
        return a

    # Recursive
    def aplusb(self, a, b):
        carry = a & b
        result = a ^ b
        return result if carry == 0 else self.aplusb(result, carry << 1)

File: bsdelf/a-b-problem.cc
/**
 * @author      Yanhui Shen
 * @version     1.0
 * @since       2016-02-11 
 */

class Solution {
public:
    /*
     * @param a: The first integer
     * @param b: The second integer
     * @return: The sum of a and b
     */
    int aplusb(int a, int b) {
        const char* c = 0;
        return reinterpret_cast<long>(&(&c[a])[b]);
    }
};